evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -0,0 +1,134 @@
+        "q": Integer(0, 10),
+    }
+    model_family = ModelFamily.ARIMA
+    supported_problem_types = [ProblemTypes.TIME_SERIES_REGRESSION]
+
+    def __init__(self, date_column=None, trend='n', p=1, d=0, q=0,
+                 random_seed=0, **kwargs):
+        order = (p, d, q)
+        parameters = {'order': order,
+                      'trend': trend}
+
+        parameters.update(kwargs)
+        self.date_column = date_column
+
+        p_error_msg = "ARIMA is not installed. Please install using `pip install statsmodels`."
+
+        arima = import_or_raise("statsmodels.tsa.arima.model", error_msg=p_error_msg)
+        try:
+            arima.ARIMA(endog=np.zeros(p + d + q + 1), **parameters)
+        except TypeError:
+            raise TypeError("Unable to instantiate ARIMA due to an unexpected argument")
+        parameters.update({'p': p,
+                           'd': d,
+                           'q': q})
+
+        super().__init__(parameters=parameters,
+                         component_obj=None,
+                         random_seed=random_seed)
+
+    def _get_dates_fit(self, X, y):
+        date_col = None
+
+        if isinstance(y.index, pd.DatetimeIndex):
+            date_col = y.index
+        if X is not None:
+            if self.date_column in X.columns:
+                date_col = X.pop(self.date_column)
+            elif isinstance(X.index, pd.DatetimeIndex):
+                date_col = X.index
+
+        if date_col is None:
+            msg = "ARIMA regressor requires input data X to have a datetime column specified by the 'date_column' parameter. " \
+                  "If not it will look for the datetime column in the index of X or y."
+            raise ValueError(msg)
+        return date_col
+
+    def _get_dates_predict(self, X, y):
+        date_col = None
+
+        if y is not None:
+            if isinstance(y.index, pd.DatetimeIndex):
+                date_col = y.index
+        if X is not None:
+            if self.date_column in X.columns:
+                date_col = X.pop(self.date_column)
+            elif isinstance(X.index, pd.DatetimeIndex):
+                date_col = X.index
+
+        if date_col is None:
+            msg = "ARIMA regressor requires input data X to have a datetime column specified by the 'date_column' parameter. " \
+                  "If not it will look for the datetime column in the index of X."
+            raise ValueError(msg)
+        return date_col
+
+    def _match_indices(self, X, y, date_col):
+        if X is not None:
+            X.index = date_col
+        if y is not None:
+            y.index = date_col
+        return X, y
+
+    def fit(self, X, y=None):
+        if y is None:
+            raise ValueError('ARIMA Regressor requires y as input.')
+
+        p_error_msg = "ARIMA is not installed. Please install using `pip install statsmodels`."
+        arima = import_or_raise("statsmodels.tsa.arima.model", error_msg=p_error_msg)
```

Comment:
Do we need this in `fit` if we have it in `__init__`?

Reply:
Complete oversight, thanks!

Reply:
We end up using it a few lines below when we initialize the ARIMA model in different ways depending on if X is passed or not :)

Reply:
Ah what I meant was, is there a case where we catch this in `fit` when we didn't catch it in `__init__`? Can a user use `fit` without init? I'm assuming the use case is:
```
arima = ARIMARegressor(...) # will catch import error
arima.fit(...) # will we ever catch import error here?
```